students/eowyn/session07/html-rendering/html_render.py

```diff
@@ -1,35 +1,32 @@
-
-
 class Element():
+    # Pretty-print HTML files with increasing indent levels
+    # for each depth of tag.
     tag = 'html'
-    indent = 4 * ' '
+    extra_indent = 4 * ' '
 
     def __init__(self, content=None):
-        # import pdb; pdb.set_trace()
         if content is None:
             self.content = []
         else:
             self.content = [content]
 
     def append(self, content):
+        # Collect additional objects alongside existing objects
         self.content.append(content)
 
-    def render(self, out_file, ind=""):
-        if self.tag == 'html':
-            out_file.write('<!DOCTYPE html>')
-        out_file.write('\n')
-        out_file.write(ind + '<' + self.tag + '>')
+    def render(self, out_file, current_ind=""):
+        # Pretty-print content and tags to file.
+        # For objects, pretty-print all the content and tags
+        # they contain to the file. Nest tags with indentation.
+
+        out_file.write('\n' + current_ind + '<' + self.tag + '>')
```

Comment:
this is going to put a newline in before every tag -- including the first one -- maybe not quite the right place?